### PR TITLE
F/wep 90 navigation menu display

### PR DESCRIPTION
**Pull request by 0815pascal**

### CODE REVIEW

File: examples/website/src/shared/navigation/navigationBar.tsx [hunk 1]
```diff
@@ -261,7 +261,9 @@ export function NavigationBar({itemsCategory, itemsIntern}: NavigationBarProps)
                 <div key={index}>
                   <Link
                     className={css(LinkStyles)}
+                    target={'_blank'}
```
Comment (roosnic1) [inline examples/website/src/shared/navigation/navigationBar.tsx:264]: `target="_blank"` only when href.

File: examples/website/src/shared/route/baseTemplateContainer.tsx [hunk 2]
```diff
@@ -37,6 +49,15 @@ export interface BaseTemplateContainerProps {
   children?: ReactNode
 }
 
+export const {
```
Comment (roosnic1) [inline examples/website/src/shared/route/baseTemplateContainer.tsx:52]: What is this?

**Review by roosnic1 — CHANGES_REQUESTED** [review]

File: examples/website/src/shared/navigation/navigationBar.tsx (hunk 1 above)
Comment (0815pascal) [reply examples/website/src/shared/navigation/navigationBar.tsx:264]: I made a distinction now between an internal link which has `target="_self"` and an external link which has `target="_blank". Can I leave it with that or should I really only use `target` when we're dealing with an external link?

Comment (roosnic1) [reply examples/website/src/shared/navigation/navigationBar.tsx]: 👍🏾  `target="self"` is bueno

File: examples/website/src/shared/route/baseTemplateContainer.tsx [hunk 3]
```diff
@@ -64,16 +74,35 @@ export function BaseTemplateContainer({
...
-  return {
-    title: link.label,
-    route: PageRoute.create({slug: link.page?.slug ?? undefined}),
-    isActive: false
+  switch (link.__typename) {
+    case 'PageNavigationLink':
+      return {
+        title: link.label,
+        route: PageRoute.create({
+          slug: link.page?.slug ?? undefined
```
Comment (roosnic1) [inline examples/website/src/shared/route/baseTemplateContainer.tsx:83]: Here is an example where you should use `!`. TypeScript doesn't know that when `link.__typename` is of type `PageNavigationLink` we have a page for sure. But we developers do. You can rewrite this line to `slug: link.page!.slug`.

File: examples/website/src/shared/route/baseTemplateContainer.tsx [hunk 4]
```diff
@@ -64,16 +74,35 @@ export function BaseTemplateContainer({
...
+        }),
+        isActive: false
+      }
+    case 'ArticleNavigationLink':
+      return {
+        title: link.label,
+        route: ArticleRoute.create({
+          id: link.article?.id ?? undefined
```
Comment (roosnic1) [inline examples/website/src/shared/route/baseTemplateContainer.tsx:91]: same as above.

**Review by roosnic1 — CHANGES_REQUESTED** [review]
Looks good. One last thing i forgot in the first review.

File: examples/website/src/shared/route/baseTemplateContainer.tsx [hunk 5]
```diff
@@ -80,15 +80,15 @@ function linkToNavigationItem(link: any): NavigationItem {
       return {
         title: link.label,
         route: PageRoute.create({
-          slug: link.page?.slug ?? undefined
+          slug: link.page!.slug ?? undefined
```
Comment (roosnic1) [inline examples/website/src/shared/route/baseTemplateContainer.tsx:83]: you don't need `?? undefined` when you use `!`

File: examples/website/src/shared/route/baseTemplateContainer.tsx [hunk 6]
```diff
@@ -80,15 +80,15 @@ function linkToNavigationItem(link: any): NavigationItem {
...
         }),
         isActive: false
       }
     case 'ArticleNavigationLink':
       return {
         title: link.label,
         route: ArticleRoute.create({
-          id: link.article?.id ?? undefined
+          id: link.article!.id ?? undefined
```
Comment (roosnic1) [inline examples/website/src/shared/route/baseTemplateContainer.tsx:91]: you don't need `?? undefined` when you use `!`

**Review by roosnic1 — CHANGES_REQUESTED** [review]

**Review by roosnic1 — APPROVED** [review]

**Review by faebebin — APPROVED** [review]
Looks good.

**Pull request merged**